pkg/cache/shares.go

```diff
@@ -84,82 +75,27 @@ func AddSharedSecret(share *sharev1alpha1.SharedSecret) {
 	br := share.Spec.SecretRef
 	key := BuildKey(br.Namespace, br.Name)
 	klog.V(4).Infof("AddSharedSecret key %s", key)
-	obj, ok := secrets.Load(key)
-	if obj != nil && ok {
-		s := obj.(*corev1.Secret)
-		secretsWithShare.Store(key, s)
+	//obj, ok := secrets.Load(key)
+	s := client.GetSecret(br.Namespace, br.Name)
+	if s != nil {
```

Comment:
Should there be an `else` that logs at least a `Warning` if this fails?

Reply:
yeah a share pointing to a  secret / configmap that does not exist is a "user error" that could emerge, where we don't want repeated error logging for it

the situation is easily discernable, in that the volume mount will not have any content, and the user can look up the share, then try to look up the shared resource and see not found

Still though ... I'll compromise and add a level 4 trace in the client.GetSecret / client.GetConfigMap calls